internal/mbsc/mbsc.go

```diff
@@ -48,3 +45,31 @@ func (m *mbsc) GetMBSC(ctx context.Context, name, namespace string) (*kmmv1beta1
 	}
 	return &mbsc, nil
 }
+
+func (m *mbsc) PatchMBSC(ctx context.Context,
+	name string,
+	namespace string,
+	moduleImageSpec *kmmv1beta1.ModuleImageSpec,
+	imageRepoSecret *v1.LocalObjectReference,
+	owner metav1.Object) error {
+	mbscObj := &kmmv1beta1.ModuleBuildSignConfig{
+		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
+	}
+
+	_, err := controllerutil.CreateOrPatch(ctx, m.client, mbscObj, func() error {
+		setModuleImageSpec(mbscObj, moduleImageSpec)
+		mbscObj.Spec.ImageRepoSecret = imageRepoSecret
+		return controllerutil.SetOwnerReference(owner, mbscObj, m.scheme)
+	})
+	return err
```

Comment:
```suggestion
	if err != nil {
	    return fmt.Errorf("failed to create or patch...", err)
	}
	
	return nil
```

Reply:
no need to add that additional error info: the code is simple, we know precisely which function produces the error, and the error itself will be self explanatory. In addition, the code that will be calling this function will append its own error message, so we will have no problem to debug it